mmedit/models/restorers/dic.py

```diff
@@ -0,0 +1,197 @@
+import numbers
+import os.path as osp
+from collections import OrderedDict
+
+import mmcv
+import torch
+
+from mmedit.core import tensor2img
+from mmedit.models.common import ImgNormalize
+from ..builder import build_backbone, build_loss
+from ..registry import MODELS
+from .basic_restorer import BasicRestorer
+
+
+@MODELS.register_module()
+class DIC(BasicRestorer):
+    """DIC model for Face Super-Resolution.
+
+    Paper: Deep Face Super-Resolution with Iterative Collaboration between
+        Attentive Recovery and Landmark Estimation.
+
+    Args:
+        generator (dict): Config for the generator.
+        pixel_loss (dict): Config for the pixel loss.
+        align_loss (dict): Config for thr align loss.
+        train_cfg (dict): Config for train. Default: None.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 pixel_loss,
+                 align_loss,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super(BasicRestorer, self).__init__()
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+        # model
+        self.generator = build_backbone(generator)
+        self.img_denormalize = ImgNormalize(
+            pixel_range=1,
+            img_mean=(0.509, 0.424, 0.378),
+            img_std=(1., 1., 1.),
+            sign=1)
+
+        # loss
+        self.pixel_loss = build_loss(pixel_loss)
+        self.align_loss = build_loss(align_loss)
+
+        # pretrained
+        if pretrained:
+            self.init_weights(pretrained)
+
+    def forward(self, lq, gt=None, test_mode=False, **kwargs):
+        """Forward function.
+
+        Args:
+            lq (Tensor): Input lq images.
+            gt (Tensor): Ground-truth image. Default: None.
+            test_mode (bool): Whether in test mode or not. Default: False.
+            kwargs (dict): Other arguments.
+        """
+
+        if test_mode:
+            return self.forward_test(lq, gt=gt, **kwargs)
+
+        return self.generator.forward(lq)
```

Comment:
It seems that this line will not be executed anyway?

Reply:
If line 93 uses `self(**data_batch, test_mode=False)`, this function can be inherited from `BasicRestorer`.

Reply:
This is not a good suggestion.
`train_step` in `BasicRestorer` includes loss and optimizer.